server/conn.go

```diff
@@ -691,18 +691,23 @@ func (cc *clientConn) Run(ctx context.Context) {
 				}
 				return
 			}
-			var txnMode string
-			if cc.ctx != nil {
-				txnMode = cc.ctx.GetSessionVars().GetReadableTxnMode()
+
+			doLogStackTrace := !shouldSkipStackTrace(err)
+			if doLogStackTrace {
```

Comment:
We should do this check in `errStrForLog()`.

Reply:
Ah didn't see that. Thanks. Just changed it.